salt/hosts-minion/init.sls

```diff
@@ -0,0 +1,14 @@
+{%- for minion_id, addrlist in salt['mine.get']('roles:kube-master', 'network.ip_addrs', 'grain').items() %}
+{{ minion_id }}-host-entry:
+  host.present:
+{% if addrlist is string %}
+    - ip: {{ addrlist }}
+{% else %}
+    - ip: {{ addrlist|first }}
+{% endif %}
+    - names:
+      - api
+      - api.{{ pillar['internal_infra_domain'] }}
+      - {{ minion_id }}
+      - {{ minion_id }}.{{ pillar['internal_infra_domain'] }}
+{% endfor %}
```

Comment:
That is a massive overstatement - this can be used in a multi master cluster, just only using one API server.

If we have enough worker nodes to DoS our API, other things in the cluster are going to fall over first.

This is not a problem, and will not be a problem for the foreseeable future, at which point a lot of how we do the system set will have changed.